src/compile/typecheck.rs

```diff
@@ -17,7 +17,7 @@ use crate::stringtable::{StringId, StringTable};
 use crate::uint256::Uint256;
 use std::collections::{HashMap, HashSet};
 
-type TypeTable<'a> = HashMap<usize, &'a Type>;
+type TypeTable<'a> = HashMap<usize, Type>;
```

Comment:
Do we really need the <`a> on TypeTable?

Reply:
Good catch, its no longer needed.